Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM python:3.11.4-buster as base
+
+RUN apt-get update
+
+ENV PATH="/root/.local/bin:$PATH"
+ENV PORT=80
+EXPOSE $PORT
+
+RUN curl -sSL https://install.python-poetry.org | python3 -
+
+WORKDIR /app/todo_app
+COPY pyproject.toml poetry.lock /app/todo_app/
+
+FROM base as production
+
+RUN poetry install --without dev
+COPY todo_app todo_app
+
+ENTRYPOINT poetry run gunicorn --bind 0.0.0.0:$PORT "todo_app.app:create_app()"
+
+FROM base as development
+
+RUN poetry install
```

Comment:
Ironically, I can get the hot-reloading working in the dev container by adding the `--without dev` here - I suspect that `pytest-watch` is switching the file-monitoring behaviour of Flask which doesn't play nicely with my Windows/WSL setup. I wouldn't actually change this, since that's likely to bite you later, but I thought worth noting!